ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuationTest.kt

```diff
@@ -0,0 +1,452 @@
+            val code =
+                """
+                val foo1 = listOf(1, 2, 3).
+                    sum()// 1
+
+                val foo2 = listOf(1, 2, 3).
+                    sum().dec()// 1
+
+                val foo3 = listOf(1, 2, 3)
+                    .filter {
+                        it > 2
+                    }
+                    .sum()// 1
+                """.trimIndent()
+            val formattedCode =
+                """
+                val foo1 = listOf(1, 2, 3)
+                    .sum()// 1
+
+                val foo2 = listOf(1, 2, 3)
+                    .sum().dec()// 1
+
+                val foo3 = listOf(1, 2, 3)
+                    .filter {
+                        it > 2
+                    }.sum()// 1
+                """.trimIndent()
+            chainMethodContinuationAssertThat(code)
+                .hasLintViolations(
+                    LintViolation(1, 28, "Expected newline before '.'"),
+                    LintViolation(4, 28, "Expected newline before '.'"),
+                    LintViolation(11, 6, "Unexpected newline before '.'"),
+                ).isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given method chain which is incorrectly wrapped with dot operator with single line comments after chain operator`() {
+            val code =
+                """
+                val foo = listOf(1, 2, 3).// 1
+                    filter {
+                        it > 2
+                    }.// 2
+                    sum()
+                """.trimIndent()
+            chainMethodContinuationAssertThat(code)
+                .hasLintViolationsWithoutAutoCorrect(
+                    LintViolation(1, 27, "Expected newline before '.'"),
+                    LintViolation(4, 7, "Unexpected newline after '}.// 2'"),
+                )
+        }
+
+        @Test
+        fun `Given method chain which is incorrectly wrapped with dot operator with single line comments and last line end with chain`() {
+            val code =
+                """
+                val foo = listOf(1, 2, 3)
+                    .filter {
+                        it > 2
+                    }.// 2
+                    sum().dec()// 3
+                """.trimIndent()
+            chainMethodContinuationAssertThat(code)
+                .hasLintViolationsWithoutAutoCorrect(
+                    LintViolation(4, 7, "Unexpected newline after '}.// 2'"),
+                )
+        }
+
+        @Test
+        fun `Given method chain which is incorrectly wrapped in next line with previous RBrace with single line comments`() {
+            val code =
+                """
+                val foo = listOf(1, 2, 3).filter {
+                    it > 2
+                }// 1
+                    .sum()// 2
+                """.trimIndent()
+            chainMethodContinuationAssertThat(code)
+                .hasLintViolationsWithoutAutoCorrect(
```

Comment:
@paul-dingemans also is this the correct way to assert that this violations are present but linter didn't do any autocorrect

Reply:
Yes, but when there is only one lint violation then use `hasLintViolationWithoutAutoCorrect` for better readability.

Reply:
I like using same method as that make me able to change the TC or copy the TC without changing the assertion based on the number of violations